include/mapbox/platform.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+// Determine compiler
+#define MB_COMPILER 0
+#define MB_COMPILER_GNU 1
+#define MB_COMPILER_CLANG 2
+#define MB_COMPILER_MSVC 3
+
+#if defined(__clang__)
+#    undef MB_COMPILER
+#    define MB_COMPILER MB_COMPILER_CLANG
+#elif defined(__GNUC__) || defined(__GNUG__) // after clang, possible it also has the same defines (need to research)
+#    undef MB_COMPILER
+#    define MB_COMPILER MB_COMPILER_GNU
+#elif defined(_MSC_VER)
+#    undef MB_COMPILER
+#    define MB_COMPILER MB_COMPILER_MSVC
+#else
+#    error "Unsupported compiler"
+#endif
+
+// Determine platform
+#define MB_PLATFORM 0
+#define MB_PLATFORM_WIN32 1
+#define MB_PLATFORM_LINUX 2
+#define MB_PLATFORM_MAC 3
+#define MB_PLATFORM_IOS_EMBEDDED 4
+#define MB_PLATFORM_IOS_SIMULATOR 5
+#define MB_PLATFORM_ANDROID 6
+#define MB_PLATFORM_QNX_NEUTRINO 7
+#define MB_PLATFORM_QNX_4 8
+
+#if (defined(__WIN32__) || defined(_WIN32)) && !defined(__ANDROID__)
+#    undef MB_PLATFORM
```

Comment:
wondering why we want to do undef here. What cases are we protecting against? 
It would be probably better to explicitly fail if there is conflict (ifdef ... error) rather than silently overriding a define.

Reply:
Removed. Compiler will generate warnings